### ISSUE TITLE
Uploading multiple large files

### ISSUE THREAD

**Issue by nikolovp** [balderdashy/skipper#116]
Hi, I have some troubles uploading multiple large files.

Skipper passes the control to the client code when one of these cases appears: the whole form is parsed; as soon as at least one file is received; timeout (50ms).

What if I want to upload multiple large files (>100KB)? Skipper will receive the first one and pass the control to my code. But the form parser isn't finished yet, req._fileparser.upstreams doesn't contain all the files from the form and I will never know about the other files?

This way Skipper is very responsive but it doesn't work the way I'm expecting (all files to be parsed when Skipper passes the control to my code).

Is there some logical explanation for the second condition "as soon as at least one file is received" or I'm missing something?

**Comment by sailsbot** [comment]
Thanks for posting, @nikolovp.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by nikolovp** [comment]
up
